docs/Golden-Gate/Triggers.rst

```diff
@@ -0,0 +1,108 @@
+
+Trigger Sinks
+*************
+Like sources, trigger sinks are boolean signals that have been annotated
+alongside their associated clock. These signals will be driven by a Boolean
+value created by the trigger system. If trigger sources exist your design, the
+generated trigger will override all assignments made in the chisel to the same
+signal, otherwise, it will take on a default value provided by the user. We
+provide an example of annotating a sink using the the ``TriggerSink`` object below.
+
+.. literalinclude:: ../../sim/src/main/scala/midasexamples/TriggerWiringModule.scala
+    :language: scala
+    :start-after: DOC include start: TriggerSink Usage
+    :end-before: DOC include end: TriggerSink Usage
+
+Trigger Timing
+---------------
+Golden Gate implements the trigger system by generating a target circuit that
+synchronizes all credit and debits into the base clock domain using a
+`single` register stage, before doing a global accounting. If the total number
+of credits exceeds debits the trigger is asserted. This trigger is then
+synchronized in each sink domain using a single register stage before driving
+the annotated sink. The circuit that implements this functionality is depicted below:
+
+.. figure:: trigger-schematic.svg
+    :width: 800px
+    :align: center
+
+    Trigger generation circuit. Not shown: a sub-circuit analagous to that
+    which totalCredit' is replicated to count debits. Similarly, the sub-circuit
+    feeding the add-reduction is generated for each clock domain that contains
+    at least one source annotation.
+
+
+Given the present implementation, an enabled trigger becomes visible in a
+sink domain no sooner than one base-clock edge and one local-clock edge have elapsed,
+in that order, after the credit was asserted. This is depicted in the
+waveform below.
+
+.. figure:: trigger-waveform.svg
+    :width: 800px
+    :align: center
+
+    Trigger timing diagram.
+
+
+Note that trigger sources and sinks that reside in the base clock domain still
+have the additional synchronization registers even though they are uneeded. Thus, a credit issued by a
+source in the base clock domain will be visible to a sink also in the base
+clock domain exactly 2 cycles after it was issued.
+
+Bridges that use the default ``HostPort`` interface add an additional cycle of
+latency in the bridge's local domain since their token channels model a single
+register stage to improve simulation FMR. Thus, without using a different
+``HostPort`` implementation, trigger sources generated by a Bridge and trigger
+sinks that feed into a Bridge will each see one additional bridge-local cycle
+of latency. In constrast, synthesized printfs and assertions, and AutoCounters
+all use wire channels (since they are unidirectional interfaces, the extra
+register stage is not required to improve FMR) and will see no additional sink
+latency.
```

Comment:
So the bottom line is that a trigger will be offset by one base-clock cycle compared to all other current debugging instrumentation?

Reply:
If i understand you correctly, yes, at least. 

in other words if you wrote:
```
val creditSig = <some bool>
val sink = <some bool>
TriggerSource.credit(creditSig)
TriggerSink(sink)
printf(synthesizePrintf(credit)))
printf(synthesizePrintf(sink)))
```
In the captured printf trace, sink would go high 2 cycles after credit went high. 

If you look at the provided waveform, you can definitely imagine situations where the trigger
reaches the sinks in certain clock domains in < 1 base clock cycle (e.g., when both the source and sink domain are faster than the base domain).